### DISCUSSION TITLE
ARC Documentation

### DISCUSSION THREAD

**Discussion by kartun83** [zepp-health/zeppos-forum#173] [Developer Feedback]
[ARC Documentation](https://docs.zepp.com/docs/1.0/reference/device-app-api/hmUI/widget/ARC/?_highlight=arc) is very confusing.

So, here is the example code of two arcs:

```
const arc = hmUI.createWidget(hmUI.widget.ARC, {
      x: 0,
      y: 0,
      radius: 25,
      w: DEVICE_WIDTH,
      h: DEVICE_HEIGHT,
      start_angle: -90,
      end_angle: 90,
      line_width: px(5),
      color: SystemColors.color_sys_warning,
    })

    const arc2 = hmUI.createWidget(hmUI.widget.ARC, {
      x: 0,
      y: 0,
      radius: 25,
      w: DEVICE_WIDTH - 20,
      h: DEVICE_HEIGHT - 20,
      start_angle: 90,
      end_angle: 270,
      line_width: px(10),
      color: SystemColors.color_sys_normal_graphic,
    })
```
So, from provided documentation it's not very clear what is X, Y and Radius.
It would be very helpful to see exact meanings of this parameters.
![Screenshot from 2023-05-15 19-44-40](https://github.com/zepp-health/zeppos-forum/assets/5894573/3301ce04-c833-4b20-908e-2baabf7a47d3)

I guess, that x and y are coordinates of top left corner of the square. It would be much more helpful to be able to define center point.

**Comment by MyoungXUE** [comment]
Hi there, 

My apologies for the confusion. First of all, you are right in your assumption. Thank you for the suggestion, I will clarify this in the documentation revision this week.

Regards,
Zepp OS Team

**Comment by MyoungXUE** [comment]
Hi, we have added this part of the drawing logic in the documentation.

https://docs.zepp.com/docs/reference/device-app-api/newAPI/ui/widget/ARC/
